libs/ledger/include/ledger/chain/main_chain.hpp

```diff
@@ -167,21 +164,22 @@ class MainChain
 
   /// @name Block Management
   /// @{
-  BlockStatus AddBlock(Block const &blk);
+  BlockStatus AddBlock(Block block);
+  BlockStatus AddBlock(BlockPtr &&block);
   BlockPtr    GetBlock(BlockHash const &hash) const;
   bool        RemoveBlock(BlockHash const &hash);
   /// @}
 
   /// @name Chain Queries
   /// @{
-  BlockPtr   GetHeaviestBlock() const;
-  BlockHash  GetHeaviestBlockHash() const;
-  Blocks     GetHeaviestChain(uint64_t limit = UPPER_BOUND) const;
-  Blocks     GetChainPreceding(BlockHash start, uint64_t limit = UPPER_BOUND) const;
-  Travelogue TimeTravel(BlockHash current_hash) const;
-  bool       GetPathToCommonAncestor(
-            Blocks &blocks, BlockHash tip_hash, BlockHash node_hash, uint64_t limit = UPPER_BOUND,
-            BehaviourWhenLimit behaviour = BehaviourWhenLimit::RETURN_MOST_RECENT) const;
+  BlockPtr      GetHeaviestBlock() const;
+  BlockHash     GetHeaviestBlockHash() const;
+  MutableBlocks GetHeaviestChain(uint64_t limit = UPPER_BOUND) const;
```

Comment:
If we are basically dropping the internal block type now, then I guess this make sense to drop the Mutable part from the name